vmm.mk

```diff
@@ -0,0 +1,70 @@
+#
+# Copyright 2024, UNSW (ABN 57 195 873 179)
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+# Snippet to build libvmm.a, to be included in a full-system Makefile.
+# Needs the variable LionsOS to point to the top of the LionsOS tree.
+#
+V3_BOARDS := BOARD_imx8mm_evk
+ifeq ($(filter ${MICROKIT_BOARD},${V3_BOARDS}),)
+	VGIC := GIC_V2
+	VGIC_FILES := src/arch/aarch64/vgic/vgic_v2.c
+else
+	VGIC := GIC_V3
+	VGIC_FILES := src/arch/aarch64/vgic/vgic_v3.c
+endif
+
+AARCH64_FILES := src/arch/aarch64/fault.c \
+		 src/arch/aarch64/linux.c \
+		 src/arch/aarch64/linux.c \
+		 src/arch/aarch64/psci.c \
+		 src/arch/aarch64/smc.c \
+		 src/arch/aarch64/tcb.c \
+		 src/arch/aarch64/vcpu.c \
+		 src/arch/aarch64/virq.c \
+		 src/arch/aarch64/vgic/vgic.c \
+		 ${VGIC_FILES}
+
+# VIRTIO MMIO depends on sddf
+ifeq ($(strip $(SDDF)),)
```

Comment:
Last thing before this can be merged:

Can you help my understand why we need 
```
ifneq (,$(wildcard ${LionsOS}/sddf))
    SDDF := ${LionsOS}/sddf
```
?

Why not just rely on the LionsOS system that is using libvmm to set the path appropriately, just like every other project would?